### PR TITLE
Fix broken debugging caused by  missing CallFrame.url

### PR DESCRIPTION
**Pull request by annagrin**
Apparently Chromium deprecated the CallFrame.url field in the debugger protocol:

https://bugs.chromium.org/p/chromium/issues/detail?id=1270316

https://chromium.googlesource.com/v8/v8/+/21f72f82ce854762d6bc1c780d3b108dc1947f5d

Use `remoteDebugger.scripts` instead of `CallFrame.url` to look up the frame url from a chrome script id.

Closes: https://github.com/dart-lang/webdev/issues/1557
Related: https://github.com/flutter/flutter/issues/101224

### CODE REVIEW

File: dwds/lib/src/services/expression_evaluator.dart [hunk 1]
```diff
@@ -321,4 +323,7 @@ class ExpressionEvaluator {
 
     return jsScope;
   }
+
+  String urlForScriptId(String scriptId) =>
```
Comment (elliette) [inline dwds/lib/src/services/expression_evaluator.dart:327]: What happens if this is `null`?

File: dwds/lib/src/debugging/debugger.dart [hunk 2]
```diff
@@ -312,15 +312,21 @@ class Debugger extends Domain {
     }
   }
 
+  String urlForScriptId(String scriptId) =>
```
Comment (natebosch) [inline dwds/lib/src/debugging/debugger.dart:315]: Can this be private?

https://dart.dev/guides/language/effective-dart/design#prefer-making-declarations-private

**Review by elliette — APPROVED** [review]

**Comment by Statyk7** [comment]
Great to see this one fixed! I'm not familiar with the process, how that fix will be distributed? Is it going to be part of the next Flutter update? If yes, anyway to get it before? Thanks!!!

**Comment by florisdipt** [comment]
Great work on the fast fix!

Could anyone let me know when this is available on (flutter/dart)stable, as currently I am unable to properly test my code and downgrading chrome is not an option for me.

Thanks

**Comment by DanielArndt** [comment]
@Statyk7 , @florisdipt You can follow from here to see the current status in Flutter as well as some steps to manually apply the fix if you need to (as of writing, merged to master, but backport for stable is still pending): https://github.com/flutter/flutter/issues/101224#issuecomment-1090830539

**Pull request merged**